gui/src-tauri/src/main.rs

```diff
@@ -237,8 +244,23 @@ fn main() {
 			WindowEvent::Resized(_) => std::thread::sleep(std::time::Duration::from_nanos(1)),
 			_ => (),
 		})
-		.run(tauri::generate_context!());
-	match run_result {
+		.build(tauri::generate_context!());
+	match build_result {
+		Ok(app) => {
+			app.run(move |_app_handle, event| match event {
+				RunEvent::ExitRequested { .. } => {
+					if let Some(mut child) = backend.0.take() {
+						let write_result = child.write(b"exit\n");
+						match write_result {
+							Ok(()) => log::info!("send exist to backend"),
+        					Err(_) => log::info!("fail to send exist to backend"),
+						}
+						thread::sleep(Duration::from_millis(10000));
```

Comment:
```suggestion
```
I would assume that sleeping is unecessary, you don't want to block the closure that is processing events.

Reply:
sleeping is necessary, since after sending exit to server. it takes some time for the server to clean up. 

If remove the sleep, the server will receive a kill signal which cleanup may not finish.

Why sleep? I think this just gives 10s to clean up or else kill the signal.

Reply:
what happens if the server cleans up before the 10s finishes?

Reply:
it just wait 10s before closing, the window is closed. User doesn't feel anything.

@ImUrX has add some comment about this. There is some limitation about windows.

Reply:
https://github.com/tauri-apps/tauri/issues/6634

Reply:
(i'm coming from the tauri issue you opened :P )
Another workaround would be to use something like an AtomicBool which you then change in your `CommandEvent::Terminated` and check for its value in ExitRequested in a loop - probably want to add a max duration as a fallback idk.

Reply:
haha, yeah seems like an idea lmao